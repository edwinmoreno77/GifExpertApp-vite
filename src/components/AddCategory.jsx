import { useState } from "react"


export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    }
    const onSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length <= 1) return;
        setCategories(categories => [inputValue, ...categories]);
        setInputValue('');
    }
    return (
        <form onSubmit={e => onSubmit(e)}>
            <input type="text"
                placeholder="Search Gif"
                value={inputValue}
                onChange={e => handleInputChange(e)} // se deja visible el evento de forma educativa

            />
        </form>
    )
}
