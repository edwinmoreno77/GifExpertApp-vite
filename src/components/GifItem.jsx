import PropTypes from 'prop-types';


export const GifItem = ({ title, url, }) => {
    return (
        <div className="card animate__animated animate__heartBeat">
            <img src={url} alt={title} />
            <p className="p"><b>{title}</b></p>
        </div>
    )
}

GifItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}