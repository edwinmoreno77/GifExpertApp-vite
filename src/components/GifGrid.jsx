import React, { useEffect, useState } from 'react'
import { getGifs } from '../helpers/getGifs';
import { GifItem } from './GifItem';





export const GifGrid = ({ cat }) => {

    const [images, setImages] = useState([]);

    const getImages = async () => {
        const newImages = await getGifs(cat);

        setImages(newImages);
    }

    useEffect(() => {
        getImages(cat);

    }, [])

    return (
        < >
            <h2>{cat}</h2>
            <div className='card-grid'>
                {
                    images.map((img) => (
                        <GifItem key={img.id} {...img} />
                    ))
                }
            </div>
        </>
    )
}