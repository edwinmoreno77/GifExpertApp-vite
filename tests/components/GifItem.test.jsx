import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";


describe('test GifItem.jsx', () => {

    const title = 'saitama';
    const url = 'https://i.gifer.com/no.gif';

    test('should render GifItem.jsx', () => {

        const { container } = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    });


    test('should render img with URL and ALT', () => {

        render(<GifItem title={title} url={url} />);
        // screen.debug();
        // expect(screen.getByRole('img').src).toBe(url);
        // expect(screen.getByRole('img').alt).toBe(title);
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);

    });

    test('should render p with title', () => {

        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();
    });

})