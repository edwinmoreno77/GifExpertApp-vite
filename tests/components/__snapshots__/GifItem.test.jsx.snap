// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test GifItem.jsx should render GifItem.jsx 1`] = `
<div>
  <div
    class="card animate__animated animate__heartBeat"
  >
    <img
      alt="saitama"
      src="https://i.gifer.com/no.gif"
    />
    <p
      class="p"
    >
      <b>
        saitama
      </b>
    </p>
  </div>
</div>
`;
